torchgeo/datamodules/potsdam.py

```diff
@@ -45,117 +41,48 @@ def __init__(
         ``num_tiles_per_batch`` x ``num_patches_per_tile``.
 
         Args:
-            num_tiles_per_batch: The number of image tiles to sample from during
-                training
-            num_patches_per_tile: The number of patches to randomly sample from each
-                image tile during training
-            patch_size: The size of each patch, either ``size`` or ``(height, width)``.
-                Should be a multiple of 32 for most segmentation architectures
-            val_split_pct: The percentage of the dataset to use as a validation set
-            num_workers: The number of workers to use for parallel data loading
+            num_tiles_per_batch: Number of image tiles to sample from during training.
+            num_patches_per_tile: Number of patches to randomly sample from each image
+                tile during training.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+                Should be a multiple of 32 for most segmentation architectures.
+            val_split_pct: Percentage of the dataset to use as a validation set.
+            num_workers: Number of workers for parallel data loading.
             **kwargs: Additional keyword arguments passed to
-                :class:`~torchgeo.datasets.Potsdam2D`
+                :class:`~torchgeo.datasets.Potsdam2D`.
 
         .. versionchanged:: 0.4
            *batch_size* was replaced by *num_tile_per_batch*, *num_patches_per_tile*,
            and *patch_size*.
         """
-        super().__init__()
+        super().__init__(Potsdam2D, 1, num_workers, **kwargs)
 
-        self.num_tiles_per_batch = num_tiles_per_batch
+        self.train_batch_size = num_tiles_per_batch
         self.num_patches_per_tile = num_patches_per_tile
         self.patch_size = _to_tuple(patch_size)
         self.val_split_pct = val_split_pct
-        self.num_workers = num_workers
-        self.kwargs = kwargs
 
-        self.train_transform = AugmentationSequential(
-            Normalize(mean=0.0, std=255.0),
+        self.train_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
             _RandomNCrop(self.patch_size, self.num_patches_per_tile),
             data_keys=["image", "mask"],
         )
-        self.test_transform = AugmentationSequential(
-            Normalize(mean=0.0, std=255.0),
+        self.aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
             _ExtractTensorPatches(self.patch_size),
```

Comment:
I am currently using this with the real dataset and observing the following: `_ExtractTensorPatches` extracts all possible patches from the scene, which if I choose a `patch_size` of 128 and given that the scene has a dimension of `4, 6000, 6000` yields a model input batch of `2209, 4, 128, 128` which is heavy on GPU memory.